pkg/kdump/config-client.js

```diff
@@ -113,21 +124,185 @@ export class ConfigFile {
+     */
+    _updateSetting(settings, key, value) {
+        if (key in settings._internal) {
+            if (value)
+                settings._internal[key].value = value;
+            else
+                delete settings._internal[key];
+        } else {
+            if (value)
+                settings._internal[key] = { value: value };
+        }
+    }
+
+    /* transform settings from model back to _internal format
+     * this.settings = current state from file
+     * settings = in-memory state from UI
+     */
+    _persistSettings(settings) {
+        // target
+        if (Object.keys(settings.targets).length > 0) {
+            const target = Object.values(settings.targets)[0];
+            this._updateSetting(settings, "path", target.path);
+
+            // wipe old target settings
+            for (const key in this.settings.targets) {
+                const oldTarget = this.settings.targets[key];
+                if (oldTarget.type == "mount") {
+                    delete settings._internal[oldTarget.fsType];
+                } else if (oldTarget.type == "ssh") {
+                    delete settings._internal.ssh;
+                    delete settings._internal.sshkey;
+                } else {
+                    delete settings._internal[key];
+                }
+            }
+
+            if (target.type === "nfs") {
+                this._updateSetting(settings, "nfs", [target.server, target.export].join(":"));
+            } else if (target.type === "ssh") {
+                this._updateSetting(settings, "ssh", target.server);
+                if ("sshkey" in target)
+                    this._updateSetting(settings, "sshkey", target.sshkey);
+            } else if (target.type === "raw") {
+                this._updateSetting(settings, "raw", target.partition);
+            } else if (target.type === "mount") {
+                this._updateSetting(settings, target.fsType, target.partition);
+            }
+
+            /* ssh target needs a flattened vmcore for transport */
+            if ("core_collector" in settings._internal &&
+                settings._internal.core_collector.value.includes("makedumpfile")) {
+                if (target.type === "ssh" && !settings._internal.core_collector.value.includes("-F"))
+                    settings._internal.core_collector.value += " -F";
+                else if (settings._internal.core_collector.value.includes("-F"))
+                    settings._internal.core_collector.value =
+                        settings._internal.core_collector.value
+                                .split(" ")
+                                .filter(e => e != "-F")
+                                .join(" ");
+            }
+        }
+        // compression
+        if (this.settings.compression.enabled != settings.compression.enabled) {
+            if (settings.compression.enabled) {
+                // enable compression
+                if ("core_collector" in settings._internal)
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17544-20220719-143653-7a0d5780-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)